### ISSUE TITLE
Interacting with Livewire From Alpine (use $wire) not working

### ISSUE THREAD

**Issue by addeye** [livewire/livewire#1478]
### Description
I tried the example interacting with Livewire From Alpine from documentation

### Steps to reproduce
![Screen Shot 2020-09-07 at 18 18 13](https://user-images.githubusercontent.com/11553632/92382639-1e2d4500-f137-11ea-968d-12575021f200.png)

![Screen Shot 2020-09-07 at 18 21 44](https://user-images.githubusercontent.com/11553632/92382651-25ece980-f137-11ea-8ecd-6df07fc800fa.png)

![Screen Shot 2020-09-07 at 18 22 00](https://user-images.githubusercontent.com/11553632/92382661-2dac8e00-f137-11ea-9595-2c1d9e8afd56.png)

**Context**
- Livewire version: 2.0
- Laravel version: 7.0
- Browser: Chrome

**Comment by nuernbergerA** [comment]
Just to be sure:
Do you pull in alpine.js?

**Comment by electronick86** [comment]
I think I have the same issue :

```
% npm list| grep alpine
├── alpinejs@2.6.0
% composer show | grep livewire
livewire/livewire                  v2.0.0    A front-end framework for Laravel.

```

My blade contains : 
```html
<button  x-on:click="$wire.delete({{$redirectServer->id}})">Delete</button>
```

And I get this error
```
VM6911:3 Uncaught (in promise) ReferenceError: $wire is not defined
    at eval (eval at saferEvalNoReturn (alpine.js?df24:1672), <anonymous>:3:21)
    at saferEvalNoReturn (alpine.js?df24:140)
    at Component.evaluateCommandExpression (alpine.js?df24:1679)
    at runListenerHandler (alpine.js?df24:878)
    at HTMLButtonElement.handler (alpine.js?df24:854)
```

Any idea?

**Comment by mblarsen** [comment]
It is not enough to npm install alpine. You all need to make sure that it is included in the JS bundle served on your page.

```js
// in one of your main scripts
import 'alpinejs'
```

You can also try to include Apline from a CDN instead, by adding a script tag to your header.

```html
<script src="https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.x.x/dist/alpine.min.js" defer></script>
```

Either should work.

**Comment by electronick86** [comment]
I have a `import 'alpinejs';` in my app.js.

Alpinejs is working, with:
```html
<div x-show="deleteInput" class="w-full flex flex-col" >
                            <input  class="my-2" placeholder="Type YES to confirm" x-model="deleteText">
                            <button  class="button rounded my-2"
                                     :class="{ 'cursor-not-allowed bg-red-100': deleteText != 'YES', 'cursor-pointer bg-red-500': deleteText === 'YES' }"
                                     x-on:click=" deleteText=== 'YES' && $wire.delete({{$redirectServer->id}})">Delete</button>
</div>
```
The class of the button change as expected when the input contains "YES". Maybe the order of the elements?

The last lines of my template are
```html 
<script src="{{ url(mix('js/app.js')) }}"></script>
@livewireScripts
</body>
</html>
```


![Laravel](https://user-images.githubusercontent.com/8201501/92389290-1bbbf280-f119-11ea-81f1-384fac41dd41.png)

**Comment by nuernbergerA** [comment]
Its a bug I'm on it.

**Comment by nuernbergerA** [comment]
I guess the problem is that the blade doesn't have any `wire` attribute

```blade
<div>
    <div x-data>
        <h1 x-text="$wire.count"></h1>

        <button x-on:click="$wire.increment()">Increment alpine</button>
    </div>
    <div>
        <h1>{{ $count }}</h1>

        <button wire:click="increment">Increment livewire</button>
    </div>
</div>
```

works fine without the its borken

**Comment by nuernbergerA** [comment]
I know its not what the docs say but this would work fine

``` blade
<div>
    <div x-data="{ count: @entangle('count') }">
        <h1 x-text="count"></h1>

        <button x-on:click="$wire.increment()">Increment</button>
    </div>
</div>
```

**Comment by nuernbergerA** [comment]
Note

```blade
<div>
    <div x-data>
        <!-- stuck on 0 and wount change -->
        <h1 x-text="$wire.count"></h1>

        <!-- send request and has a proper payload -->
        <button x-on:click="$wire.increment()">$wire.increment()</button>
    </div>
    <div x-data>
        <!-- will update only if the button below is click, value from server +1 -->
        <h1 x-text="$wire.count"></h1>

        <!-- send request and has a proper payload -->
        <button x-on:click="$wire.count++">$wire.count++</button>
    </div>
</div>
```

**Comment by addeye** [comment]
> I know its not what the docs say but this would work fine
> 
> ```
> <div>
>     <div x-data="{ count: @entangle('count') }">
>         <h1 x-text="count"></h1>
> 
>         <button x-on:click="$wire.increment()">Increment</button>
>     </div>
> </div>
> ```

it's working.

**Comment by calebporzio** [comment]
Thank you for submitting this.

The issue was caused by the new payload optimization that sometimes doesn't send HTML down the wire.

In your case, no HTML was being sent from Livewire because nothing changed on the PHP side, so Alpine never had the chance to update.

This is all fixed and tagged in V2.0.1

Thanks again everyone!

**Comment by RahulDey12** [comment]
@calebporzio 
Still facing this issue 
![image](https://user-images.githubusercontent.com/26340814/92648443-0ad0d400-f307-11ea-94ab-4bb1f7d406f8.png)
![image](https://user-images.githubusercontent.com/26340814/92648499-1e7c3a80-f307-11ea-8b42-b931d52b15d4.png)
![image](https://user-images.githubusercontent.com/26340814/92648553-35229180-f307-11ea-9820-3ea065356516.png)

**Comment by Haizone** [comment]
stil didnt work
 "livewire/livewire": "^2.2",

i get:

`Cannot read property '$wire' of undefined`

**Comment by nuernbergerA** [comment]
@Haizone and @RahulDey12 
please open a new issue with domain-free code examples to reproduce the issue,

thanks

**Comment by addeye** [comment]
> stil didnt work
> "livewire/livewire": "^2.2",
> 
> i get:
> 
> `Cannot read property '$wire' of undefined`

solved using alpine v2.6.0
